packages/tools/kda-cli/src/hooks/use-history.ts

```diff
@@ -0,0 +1,75 @@
+import fs from 'fs';
+import { useCallback, useState } from 'react';
+const readOrCreateHistory = (): Record<string, HistoryValue> => {
+  try {
+    return JSON.parse(fs.readFileSync('.kda-history.json', 'utf-8'));
+  } catch (e) {
+    return {};
+  }
+};
+
+export type HistoryValue =
+  | string
+  | boolean
+  | string[]
+  | Record<string, unknown>;
+
+const isSuggestiable = (value: unknown): value is string[] => {
+  return Array.isArray(value);
+};
+
+type Callback = (newValue: HistoryValue) => void;
+export const useHistory = (
+  name: string,
+): { history: HistoryValue; onSave: Callback; onSet: Callback } => {
+  const [history, setHistory] = useState<Record<string, HistoryValue>>(
+    readOrCreateHistory(),
+  );
+  const onSave = useCallback(
+    (newValues: HistoryValue): void => {
+      const h = readOrCreateHistory();
+      const v = Array.isArray(newValues) ? newValues : [newValues];
+      const readSuggestions = h[name];
+      const suggestions: string[] = isSuggestiable(readSuggestions)
+        ? readSuggestions
+        : [];
+      fs.writeFileSync(
+        '.kda-history.json',
+        JSON.stringify(
+          {
+            ...h,
+            [name]: Array.from(new Set([...suggestions, ...v]).values()),
+          },
+          null,
+          2,
+        ),
+        'utf-8',
+      );
+    },
+    [history, setHistory, name],
+  );
+  const onSet = useCallback(
+    (newValue: HistoryValue): void => {
+      const h = readOrCreateHistory();
+      fs.writeFileSync(
+        '.kda-history.json',
+        JSON.stringify(
+          {
+            ...h,
+            [name]: newValue,
+          },
+          null,
+          2,
+        ),
+        'utf-8',
+      );
+    },
+    [history, setHistory, name],
```

Comment:
Does it write the file to the package directory, or the `process.cwd()` or a user dir (e.g. `~/.kda-history.json`)?

Reply:
I deliberately let it write to the current cwd, as the cli tooling will be having a different relevant history for each project it's assisting with. (different local calls, deployments etc)